pkgs/tools/misc/coreutils/default.nix

```diff
@@ -138,7 +138,10 @@ stdenv.mkDerivation rec {
     # the boot time is set to the epoch because the system has no RTC. We
     # explicitly enable it for cases where it can't be detected automatically,
     # such as when cross-compiling.
-    ++ optional stdenv.hostPlatform.isLinux "gl_cv_have_proc_uptime=yes";
+    ++ optional stdenv.hostPlatform.isLinux "gl_cv_have_proc_uptime=yes"
+    # this function exists in libc but is not provided by any header, leading
+    # to its detection at configure flag and failure at build time, I think
+    ++ optional stdenv.hostPlatform.isFreeBSD "ac_cv_func_random_r=no";
```

Comment:
No, on non-Nix builds this value is detected automatically. I'm really not sure what could cause this

Reply:
Might be worth comparing config.log between Nix and non-Nix builds?

Reply:
Without a doubt, chasing this rabbit hole was the most awful experience I've had debugging freebsd/nix so far. Our libc had a bunch of local symbols in it, which are usually removed by stripping. These are what was being picked up by coreutils configure to believe that random_r existed. The reason they weren't being stripped was that *our install wrapper purposefully discarded the flag that would cause it to invoke strip*. The nixpkgs blame goes back to the original init freebsd commit. I can't for the life of me figure out why this would be considering that the same commit specifies that that flag should be passed! I suppose it has fallen to me to debug this weighty error. Everyone say, "Thank you, Atlas". Fix pushed.

Reply:
Thank you so much for the wonderful investigation!